### PR TITLE
Support separate worlds, keyed by the root feature ID.

### PR DESCRIPTION
**Pull request by andreweland**
Support separate worlds, using the ID that's currently used for the feature that configures the UI to select which world. Support passing this ID through Python, and add it to the tile requests from javascript. TODO: this ID is currently called "context", "root", or "UI context", we should standardise on something.

### CODE REVIEW

File: src/diagonal.works/b6/ingest/worlds.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"diagonal.works/b6"
+)
+
+var DefaultWorldFeatureID = b6.FeatureID{
+	Type:      b6.FeatureTypeCollection,
+	Namespace: "diagonal.works/world",
+	Value:     0,
+}
+
+type Worlds interface {
```
Comment (xprogramari) [inline src/diagonal.works/b6/ingest/worlds.go:13]: should this maybe live in mutable / and be mutable worlds ?

File: src/diagonal.works/b6/ingest/worlds.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	FindWorld(id b6.FeatureID) MutableWorld
+}
+
+type MutableWorlds struct {
```
Comment (xprogramari) [inline src/diagonal.works/b6/ingest/worlds.go:17]: maybe this could all be in mutable ?

File: src/diagonal.works/b6/ingest/worlds.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	Base    b6.World
+	Mutable map[b6.FeatureID]MutableWorld
+}
+
+func (m *MutableWorlds) FindWorld(id b6.FeatureID) MutableWorld {
+	if !id.IsValid() {
+		id = DefaultWorldFeatureID
```
Comment (xprogramari) [inline src/diagonal.works/b6/ingest/worlds.go:24]: i m not sure i like this default id and fallback / y do we want this ?

File: src/diagonal.works/b6/ingest/worlds.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	}
+	w, ok := m.Mutable[id]
+	if !ok {
+		if m.Mutable == nil {
+			m.Mutable = make(map[b6.FeatureID]MutableWorld)
```
Comment (xprogramari) [inline src/diagonal.works/b6/ingest/worlds.go:29]: y :( y create in find

**Review by xprogramari — APPROVED** [review]

File: src/diagonal.works/b6/ingest/worlds.go (hunk 2 above)
Comment (andreweland) [reply src/diagonal.works/b6/ingest/worlds.go:17]: I think I'll keep all the Worlds code together for the moment, and maybe split it out later when we better know the shape.

File: src/diagonal.works/b6/ingest/worlds.go (hunk 3 above)
Comment (andreweland) [reply src/diagonal.works/b6/ingest/worlds.go:24]: I don't know whether we'll have it forever - but right now, we have a --world command line argument, so effectively this ID maps to that world. In future, we may end up with a config file mapping IDs to specific indices, but I'm not sure what that will look like yet, nor whether it'll live in the open source codebase or not, so I'm sticking with this simple approach for the moment.

File: src/diagonal.works/b6/ingest/worlds.go (hunk 4 above)
Comment (andreweland) [reply src/diagonal.works/b6/ingest/worlds.go:29]: Renamed FindOrCreateWorld. I don't want to separate for the moment, as I'd have to add functions to check if a world exists, etc.

**Pull request merged**